File: app/src/main/java/hm/orz/chaos114/android/tumekyouen/model/StageCountModel.java
package hm.orz.chaos114.android.tumekyouen.model;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class StageCountModel {

    public abstract int stageCount();

    public abstract int clearStageCount();

    public static StageCountModel create(int stageCount, int clearStageCount) {
        return new AutoValue_StageCountModel(stageCount, clearStageCount);
    }
}

File: app/src/main/java/hm/orz/chaos114/android/tumekyouen/model/StageCountModel.kt
package hm.orz.chaos114.android.tumekyouen.model

import com.google.auto.value.AutoValue

@AutoValue
abstract class StageCountModel {

    abstract fun stageCount(): Int

    abstract fun clearStageCount(): Int

    companion object {

        fun create(stageCount: Int, clearStageCount: Int): StageCountModel {
            return AutoValue_StageCountModel(stageCount, clearStageCount)
        }
    }
}
